### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by dhanasp**
Update documentation for 
- Agents API v6
- Environment Config API v3

### CODE REVIEW

**Comment by dhanasp** [comment]
Description of Environment Origin Type - 

Screenshot 2019-09-17 at 12 25 56 PM

Description of environment on bulk update and update call- 

Screenshot 2019-09-17 at 12 26 07 PM

File: source/includes/agents/_30-update.md.erb [hunk 1]
```diff
@@ -77,7 +77,7 @@ Atleast one of the following properties must be specified, not specifying the pr
   describe_object(nil) do
     hostname              'String',             'The new hostname'
     resources             ['String', 'Array'],  'A comma separated strings of resources, or an array of string resources. Updating of resources is **NOT** supported by elastic agents.'
-    environments          ['String', 'Array'],  'A comma separated strings of environments, or an array of string environments'
+    environments          ['String', 'Array'],  'A comma separated strings of environments, or an array of string environments. If the given enviornment is present, the agent will be mapped to it.'
```
Comment (maheshp) [inline source/includes/agents/_30-update.md.erb]: How about,
`A comma separated strings of environments, or an array of string environments. The agent will be associated to an environment only if an environment exists in GoCD for the given name, else will be ignored.`

File: source/includes/agents/_00-describe.md.erb [hunk 2]
```diff
@@ -57,6 +57,10 @@ end
 
 <%=
 describe_sub_object 'The Environment Origin object', json: origin_json, html_id: 'the-agent-environments-origin-object' do
-  type             'String',  'The origin type of the environment.', one_of: %w(gocd config-repo)
+  type             'String',  'The origin type of the environment.', list: [
+  '<strong>gocd</strong> : The environment is defined in <code>cruise-config.xml</code>',
```
Comment (maheshp) [inline source/includes/agents/_00-describe.md.erb]: The environment is defined in GoCD <code>cruise-config.xml</code>',

File: source/includes/agents/_00-describe.md.erb [hunk 3]
```diff
@@ -57,6 +57,10 @@ end
...
+  '<strong>config-repo</strong>: The environment is defined via <code>config-repo</code>',
```
Comment (maheshp) [inline source/includes/agents/_00-describe.md.erb]: The environment is defined via an external <code>config-repo</code>',

File: source/includes/changelog/_19-09-0.md.erb [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
\ No newline at end of file
+### Added
+
+- Introduced version 3 of [Environment Config API](#environment-config)
+- Introduced version 6 of [Agents API](#agents)
+
+### Removed
+
+- <%= removed_api_message({api_name: 'Environment Config', old_api_version: 2, new_api_version: 3, api_section: 'environment-config'}) %>
+- <%= removed_api_message({api_name: 'Agents', old_api_version: 4, new_api_version: 6, api_section: 'agents'}) %>
```
Comment (maheshp) [inline source/includes/changelog/_19-09-0.md.erb]: Should we add a note for removal without deprecation?

Comment (kritika-singh3) [reply source/includes/changelog/_19-09-0.md.erb]: Makes sense. Should we link it to the agents PR?

Comment (maheshp) [reply source/includes/changelog/_19-09-0.md.erb]: We can.

Comment (kritika-singh3) [reply source/includes/changelog/_19-09-0.md.erb]: Updated the changelog.

**Pull request merged**